addons/account/models/account_move.py

```diff
@@ -4039,12 +4039,29 @@ def _get_pdf_and_send_invoice_vals(self, template):
             'checkbox_download': False,
         }
 
-    def _generate_pdf_and_send_invoice(self, template, from_cron=True):
+    def _generate_pdf_and_send_invoice(self, template, from_cron=True, allow_invoice_proforma=True):
+        """ Generate the pdf for the current invoices and send them by mail using the send & print wizard.
+
+        :param from_cron:   Flag indicating if the method is called from a cron. In that case, we avoid raising any
+                            error.
+        :param allow_invoice_proforma:  In case of error when generating the documents for invoices, generate a
+                                        proforma PDF report instead.
+        """
         composer_vals = self._get_pdf_and_send_invoice_vals(template)
         composer = self.env['account.move.send'].create(composer_vals)
 
         # from_cron=True to log errors in chatter instead of raise
-        composer.action_send_and_print(from_cron=from_cron)
+        composer.action_send_and_print(from_cron=from_cron, allow_invoice_proforma=allow_invoice_proforma)
+
+    def _get_invoice_pdf_report_filename(self):
+        """ Get the filename of the generated PDF invoice report. """
+        self.ensure_one()
+        return f"{self.name.replace('/', '_')}.pdf"
+
+    def _get_invoice_proforma_pdf_report_filename(self):
+        """ Get the filename of the generated proforma PDF invoice report. """
+        self.ensure_one()
+        return f"{self.name.replace('/', '_')}_proforma.pdf"
```

Comment:
These 2 methods are almost identical.
Couldn't we use the same one with a parameter?

Reply:
I'll keep both separated because `_get_invoice_pdf_report_filename` will be overridden in a lot of localizations